### DISCUSSION TITLE
Masterchef 👨‍🍳

### DISCUSSION THREAD

**Discussion by CosasDePuma** [xyflow/xyflow#1197] [Show and tell]
👋 to everyone,

I am currently trying to design an application that allows to perform cybersecurity audits in a visual way.

It's called [Masterchef](https://github.com/cosasdepuma/masterchef) 🍳

![screenshot](https://user-images.githubusercontent.com/26680023/118334721-6d214a80-b50e-11eb-91d2-0683dbf42fcd.png)

The idea was born because currently there are very good tools like [Cyberchef](https://gchq.github.io/CyberChef/) that allow you to generate "recipes" with the steps you need, and I wanted to create my own focused on web pentesting.

At first, I used a design similar to [Cyberchef](https://gchq.github.io/CyberChef/), but I discovered that it was very limited because everything was so linear.

That's how I decided to try my luck with ReactFlow. It solves a lot of problems and the result is being similar to what I have in mind.

Now I have encountered a problem that I have not been able to solve: I am not able to extrapolate the execution order of the nodes. The idea is to achieve something similar to what is shown in [this video](https://www.youtube.com/watch?v=WjJdaDXN5Vs) or [this program that also uses ReactFlow](https://datablocks.pro/).

The truth is that this is my first serious project with web interface (and specifically with React ⚛️). If anyone would like to collaborate or give an opinion about it, any comment is welcome. 🤗

Hopefully I can finish it and be able to present it as a fully functional tool 💞.

**Comment by moklick** [comment]
Hey @CosasDePuma 

this looks very interesting! I saw Cyberchef before and I think Masterchef is a great advancement :)
